### ISSUE TITLE
ESLint complains about unused variable when variable of it is used to define Typescript parameter type

### ISSUE THREAD

**Issue by justin13888** [eslint/eslint#13263]
**Tell us about your environment**

* **ESLint Version:** 6.8.0
* **Node Version:** 12.16.3
* **npm Version:** 6.14.5

**What parser (default, Babel-ESLint, etc.) are you using?**
@typescript-eslint/parser
**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
    "env": {
        "es6": true,
        "node": true
    },
    "extends": [
        "google"
    ],
    "globals": {
        "Atomics": "readonly",
        "SharedArrayBuffer": "readonly"
    },
    "parser": "@typescript-eslint/parser",
    "parserOptions": {
        "ecmaVersion": 2018,
        "sourceType": "module"
    },
    "plugins": [
        "@typescript-eslint"
    ],
    "rules": {
        "indent": [
            "error",
            4
        ],
        "require-jsdoc": 0
    }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
import * as Discord from 'discord.js';
export class CommandService {
    parseCommand(msg: Discord.Message) {
        // if (msg instanceof Discord.Message) console.log('');
    }
}

```

```bash
eslint "**/*.ts"
```

**What did you expect to happen?**
Should have no error.

**What actually happened? Please include the actual, raw output from ESLint.**
services/command.service.ts
  1:13  error  'Discord' is defined but never used  no-unused-vars

✖ 1 problem (1 error, 0 warnings)

**Are you willing to submit a pull request to fix this bug?**
Don't know how.

**Comment by anikethsaha** [comment]
You need to extend [typescript-eslint](https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/eslint-plugin#usage) rules if you are using it for typescript. 
for `no-unused-vars`, use [this rule](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-unused-vars.md) and [`no-unused-vars-experimental`](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-unused-vars-experimental.md).

**Comment by justin13888** [comment]
Thanks. Really hope they implement the work around soon.

**Comment by anikethsaha** [comment]
> Really hope they implement the work around soon.

eslint supports typescript through `typescript-eslint` project. So I guess it is expected behavior.

**Comment by justin13888** [comment]
The code you mentioned isn't automatically generated when using `eslint --init`. So others may easily face this issue as well in the future.

**Comment by mdjermanovic** [comment]
> The code you mentioned isn't automatically generated when using `eslint --init`. So others may easily face this issue as well in the future.

This should be fixed with #13235
